sys/net/include/netdev.h

```diff
@@ -0,0 +1,271 @@
+     *          values to may define higher values, but they must be greater
+     *          or equal to @ref NETDEV_OPT_LAST.
+     */
+    NETDEV_OPT_LAST,
+} netdev_opt_t;
+
+/**
+ * @brief   Definition of basic network device.
+ * @note    Feel free to expand if your device needs/supports more
+ */
+typedef enum {
+    NETDEV_ACTION_POWEROFF = 0,
+    NETDEV_ACTION_SLEEP,
+    NETDEV_ACTION_IDLE,
+    NETDEV_ACTION_RX,
+    NETDEV_ACTION_TX,               /**< only needed if pre-loading is enabled */
+} netdev_action_t;
+
+/**
+ * TODO: document...
+ */
+typedef enum {
+    NETDEV_EVENT_RX_STARTED = 0,
+    NETDEV_EVENT_RX_COMPLETE,
+    NETDEV_EVENT_TX_STARTED,
+    NETDEV_EVENT_TX_COMPLETE
+    /* probably more? */
+} netdev_event_t;
+
+/**
+ * @brief   Definition of basic network device.
+ * @note    Feel free to expand if your device needs/supports more
+ */
+typedef enum {
+    NETDEV_STATE_OFF = 0,           /**< Device is powered off */
+    NETDEV_STATE_SLEEP,             /**< Device is sleeping */
+    NETDEV_STATE_IDLE,              /**< Device is idle */
+    NETDEV_STATE_RX,                /**< Device is in receive mode */
+    NETDEV_STATE_TX,                /**< Device is transmit mode */
+    NETDEV_STATE_PROMISCUOUS,       /**< Device is in receive mode and
+                                         accepts all packets without regard
+                                         for their destination */
+    NETDEV_STATE_TX_BURST,          /**< Device is burst sending and
+                                         does not accept packets */
+} netdev_state_t;
+
+/**
+ * @brief   The netdev data-structure holds the minimum information needed for
+ *          interaction with MAC layers
+ *
+ * The netdev structure is the parent for all network device driver descriptors.
+ */
+typedef struct netdev_t netdev_t;
+
+/**
+ * @brief   Event callback for signaling event to a MAC layer.
+ *
+ * @param[in] type          type of the event
+ * @param[in] arg           event argument, can e.g. contain a pkt_t pointer
+ */
+typedef void (*netdev_event_cb_t)(netdev_event_t type, void *arg);
+
+/**
+ * @brief   Network device API definition.
+ *
+ * @details This is a set of functions that must be implemented by any driver
+ *           for a network device.
+ */
+typedef struct {
+    /**
+     * @brief Send data via a given network device
+     *
+     * @param[in] dev               the network device
+     * @param[in] pkt              pointer to the data to sent
+     *
+     * @return  the number of byte actually send on success
+     * @return  -EAFNOSUPPORT if address of length dest_len is not supported
+     *          by the device *dev*
+     * @return  -ENODEV if *dev* is not recognized
+     * @return  -EMSGSIZE if the total frame size is too long to fit in a frame
+     *          of the device *dev*
+     * @return  a fitting negative other errno on other failure
+     */
+    int (*send_data)(netdev_t *dev, pkt_t *pkt);
```

Comment:
We need some address data here. (the original had `dest` and `dest_len` but I was thinking that it might be a good idea to include `src` and `src_len` too now that #2286 makes send and receive commands symmetrical.